flink-python/src/main/java/org/apache/flink/table/runtime/typeutils/serializers/python/DateSerializer.java

```diff
@@ -39,6 +41,11 @@
 
     public static final DateSerializer INSTANCE = new DateSerializer();
 
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private static final DataStructureConverter<Integer, Date> converter =
```

Comment:
we don't need to go through `DataStructureConverter`. we can use `DataTimeUtils` directly?

Reply:
My intention here was to hide the usage of `DateTimeUtils` completely. Also this class itself seems like a wrapper of a `DataStructureConverter`

Reply:
@slinkydeveloper you forgot to drop the static field. This `DateSerializer` is actually technical debt because the Python API still uses the old type system and thus has to use `java.sql.` types. Hopefully this will be fixed at some point.